jcl/src/java.base/share/classes/java/lang/invoke/ArrayVarHandle.java

```diff
@@ -163,66 +150,69 @@ private static final IllegalArgumentException newIllegalArgumentException(Class<
 		}
 
 		static final class OpObject extends ArrayVarHandleOperations {
+			private static final int OFFSET = _unsafe.arrayBaseOffset(Object[].class);
+			private static final int SCALE = _unsafe.arrayIndexScale(Object[].class);
+			
 			private static final Object get(Object receiver, int index, ArrayVarHandle varHandle) {
 				receiver.getClass();
 				boundsCheck(((Object[])receiver).length, index);
-				return _unsafe.getObject(receiver, varHandle.computeIndex(index));
+				return _unsafe.getObject(receiver, computeIndex(index, OFFSET, SCALE));
```

Comment:
Perhaps it would improve clarity and avoid the possibility of permuting the arguments to 'computeIndex' by introducing a single-argument version that mixes in OFFSET and SCALE.

Reply:
I was hoping to achieve this, but it would mean copying the `computeOffset` method body into every type-specific subclass.

Reply:
Fixed